docs/en/observability/monitor-java-app.asciidoc

```diff
@@ -0,0 +1,2005 @@
+using the admin user, and then use an API key as well.
++
+[source,console]
+----
+POST /_security/api_key
+{
+  "name": "metricbeat_javalin-app",
+  "role_descriptors": {
+    "metricbeat_writer": {
+      "cluster": ["monitor", "read_ilm"],
+      "index": [
+        {
+          "names": ["metricbeat-*"],
+          "privileges": ["view_index_metadata", "create_doc"]
+        }
+      ]
+    }
+  }
+}
+----
++
+Don’t forget to do the initial setup like this.
++
+[source,bash]
+----
+./metricbeat setup -e -E 'cloud.id=${CLOUD_ID}' -E 'cloud.auth=elastic:YOUR_SUPER_SECRET_PASS'
+----
++
+. Store the combination of `id` and `api_key` fields in the keystore.
++
+[source,bash]
+----
+./metricbeat keystore create
+echo -n "IhrJJHMB4JmIUAPLuM35:1GbfxhkMT8COBB4JWY3pvQ" | ./metricbeat keystore add ES_API_KEY --stdin
+echo -n "observability-javalin-app:ZXUtY2VudHJhbC0xLmF3cy5jbG91ZC5lcy5pbyQ4NDU5M2I1YmQzYTY0N2NhYjA2MWQ3NTJhZWFhNWEzYyQzYmQwMWE2OTQ2MmQ0N2ExYjdhYTkwMzI0YjJiOTMyYQ==" | ./metricbeat keystore add CLOUD_ID --stdin
+----
++
+. Configure {metricbeat} to read our Prometheus metrics. Start with a
+basic `metricbeat.yaml`.
++
+[source,yaml]
+----
+metricbeat.config.modules:
+  path: ${path.config}/modules.d/\*.yml
+  reload.enabled: false
+
+name: javalin-metrics-shipper
+
+cloud.id: ${CLOUD_ID}
+output.elasticsearch:
+  api_key: ${ES_API_KEY}
+
+processors:
+  - add_host_metadata: ~
+  - add_cloud_metadata: ~
+  - add_docker_metadata: ~
+  - add_kubernetes_metadata: ~
+----
++
+As {metricbeat} supports dozens of modules, which in turn are different
+ways of ingesting metrics (the same applies to {filebeat} with different
+types of log files and formats), the Prometheus module needs to be enabled.
++
+[source,bash]
+----
+./metricbeat modules enable prometheus
+----
++
+Add the Prometheus endpoint to poll.
+`./modules.d/prometheus.yml`:
++
+[source,yaml]
+----
+- module: prometheus
+  period: 10s
+  hosts: ["localhost:7000"]
+  metrics_path: /metrics
+  username: "metrics"
+  password: "secret"
+  use_types: true
+  rate_counters: true
+----
++
+To improve security, add the username and the
+password to the keystore and refer it here as well.
++
+. Start {metricbeat} and verify that the Prometheus events are flowing into
+{es}.
++
+[source,console]
+----
+GET metricbeat-7.9.0/_search?filter_path=**.prometheus,hits.total
+{
+  "query": {
+    "term": {
+      "event.module": "prometheus"
+    }
+  }
+}
+----
+
+[discrete]
+== Step 5: View metrics in {kib}
+
+As this is custom data from our Javalin app, there is no pre-defined
+dashboard for displaying this data.
+
+Let’s check for the number of logging messages per log level.
+
+[source,console]
+----
+GET metricbeat-7.9.0/_search
+{
+  "query": {
+    "exists": {
+      "field": "prometheus.log4j2_events_total.counter"
+    }
+  }
+}
+----
+
+Visualize the number of log messages over time, split by the
+log level. Since the Elastic Stack 7.7, there is a new way of creating a
```

Comment:
I recommend mentioning versions or describing a feature as a "new way of creating" things. This kind of information gets stale very quickly and is likely to stay around longer than you want it to.